packages/cli/src/metadataGeneration/extension.ts

```diff
@@ -32,3 +30,34 @@ export function getExtensions(decorators: ts.Identifier[], metadataGenerator: Me
 
   return extensions;
 }
+
+export function getExtensionsFromJSDocComments(comments: string[]): Tsoa.Extension[] {
+  const extensions: Tsoa.Extension[] = [];
+  comments.forEach(comment => {
+    const extensionData = safeFromJson(comment);
+    if (extensionData) {
+      const keys = Object.keys(extensionData);
+      keys.forEach(key => {
+        validateExtensionKey(key);
+
+        extensions.push({ key: key, value: extensionData[key] });
+      });
+    }
+  });
+
+  return extensions;
+}
+
+function validateExtensionKey(key: string) {
+  if (key.indexOf('x-') !== 0) {
+    throw new Error('Extensions must begin with "x-" to be valid. Please see the following link for more information: https://swagger.io/docs/specification/openapi-extensions/');
+  }
+}
+
+function safeFromJson(json: string) {
```

Comment:
Can you move this to utils and reuse it [here](https://github.com/lukeautry/tsoa/blob/master/packages/cli/src/metadataGeneration/typeResolver.ts#L1091)